### PR TITLE
ICMSLST-647 Fix non-bugs found by mypy.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Other changes:
  - enable check_untyped_defs flag
  - exclude migrations from mypy checks for speed

### CODE REVIEW

File: web/models/mixins.py [hunk 1]
```diff
@@ -4,23 +4,23 @@
...
     # e.g. the system shouldn't allow you to have more than one active
     # translation for language X for some template (see what was done in
     # ICMSLST-483)
     def unarchive(self):
         self.is_active = True
-        self.save()
+        self.save()  # type: ignore[attr-defined]
 
 
 class Sortable(object):
```
Comment (kevinetienne) [inline web/models/mixins.py:18]: I think we can get rid of this class. We are inheriting from `Sortable` in https://github.com/uktrade/icms/blob/901a456733a734127d206af5473e6d62b740c52a/web/domains/firearms/models.py#L89

But `swap_order` is not used.

Comment (kevinetienne) [reply web/models/mixins.py:18]: Happy to create a ticket for it let me know.

Comment (kevinetienne) [reply web/models/mixins.py:18]: Created ICMSLST-700

**Review by kevinetienne — APPROVED** [review]

File: web/domains/country/views.py [hunk 2]
```diff
@@ -221,6 +227,11 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class MissingTranslationsDict(TypedDict):
```
Comment (oskusalerma) [inline web/domains/country/views.py:230]: all types at top of files

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/country/views.py (hunk 2 above)
Comment (MattHolmes123) [reply web/domains/country/views.py]: Done

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**